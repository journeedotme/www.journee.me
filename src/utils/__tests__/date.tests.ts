import {
  createDailyInterval,
  createDailyIntervalWithDayNumberAndName,
} from "../date"

describe("date test suite", () => {
  it("should return all dates from today", () => {
    jest.useFakeTimers("modern")

    jest.setSystemTime(new Date("2020-01-03").getTime())

    const dates = createDailyInterval(7)

    expect(dates).toMatchSnapshot()
  })

  it("should return all dates from today with name and number", () => {
    jest.useFakeTimers("modern")

    jest.setSystemTime(new Date("2020-01-03").getTime())

    const dates = createDailyIntervalWithDayNumberAndName(7)

    expect(dates).toMatchSnapshot()
  })
})
