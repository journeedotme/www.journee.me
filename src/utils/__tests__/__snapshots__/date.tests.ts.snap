// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date test suite should return all dates from today 1`] = `
Array [
  "2019-12-28",
  "2019-12-29",
  "2019-12-30",
  "2019-12-31",
  "2020-01-01",
  "2020-01-02",
  "2020-01-03",
]
`;

exports[`date test suite should return all dates from today with name and number 1`] = `
Array [
  Object {
    "day": "Sat",
    "number": "28",
  },
  Object {
    "day": "Sun",
    "number": "29",
  },
  Object {
    "day": "Mon",
    "number": "30",
  },
  Object {
    "day": "Tue",
    "number": "31",
  },
  Object {
    "day": "Wed",
    "number": "1",
  },
  Object {
    "day": "Thu",
    "number": "2",
  },
  Object {
    "day": "Fri",
    "number": "3",
  },
]
`;
